src/webui/www/private/scripts/pathAutofill.js

```diff
@@ -0,0 +1,97 @@
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+"use strict";
+
+/*
+File implementing auto-fill for the path input field in the path dialogs.
+*/
+if (window.qBittorrent === undefined)
+    window.qBittorrent = {};
+
+if (window.qBittorrent.pathAutofill === undefined) {
+    window.qBittorrent.pathAutofill = (() => {
+        const exports = () => {
+            return {
+                attachPathAutofill: attachPathAutofill
+            };
+        };
+
+        function showInputSuggestions(inputElement, names) {
+            const datalist = document.createElement("datalist");
+            datalist.id = inputElement.id + "Suggestions";
+            for (const name of names) {
+                const option = document.createElement("option");
+                option.value = name;
+                datalist.appendChild(option);
+            }
+
+            const oldDatalist = document.getElementById(`${inputElement.id}Suggestions`);
+            if (oldDatalist !== null) {
+                oldDatalist.replaceWith(datalist);
+            }
+            else {
+                inputElement.appendChild(datalist);
+                inputElement.setAttribute("list", datalist.id);
+            }
+        }
+
+        function showPathSuggestions(element, mode) {
+            const partialPath = element.value;
+            if (partialPath === "")
+                return;
+
+            fetch(`api/v2/app/getDirectoryContent?dirPath=${partialPath}&mode=${mode}`)
+                .then(response => response.json())
+                .then(filesList => { showInputSuggestions(element, filesList); })
+                .catch(error => {});
+        }
+
+        // get all input text fields with class pathDirectory class="pathDirectory"
+        function attachPathAutofill() {
+            const directoryInputs = document.querySelectorAll(".pathDirectory:not(.autoFill)");
+            for (const input of directoryInputs) {
+                input.addEventListener("input", function() { showPathSuggestions(this, "dirs"); });
+                input.classList.add("autoFill");
+            }
+
+            const fileInputs = document.querySelectorAll(".pathFile:not(.autoFill)");
+            for (const input of fileInputs) {
+                input.addEventListener("input", function() { showPathSuggestions(this, "all"); });
+                input.classList.add("autoFill");
+            }
+
+            if ((directoryInputs.length === 0) && (fileInputs.length === 0))
+                console.warn("No matching elements for path autocompletion found.");
```

Comment:
When it is possible to run the scan multiple times, it would eventually spew this message and it would be confusing when it happens.
It is only helpful in debugging stage and should be removed in production.

Reply:
You convinced me